src/client/dfs/dfs.c

```diff
@@ -7939,3 +7939,128 @@ dfs_readdir_with_filter(dfs_t *dfs, dfs_obj_t *obj, dfs_pipeline_t *dpipe, daos_
 	D_FREE(buf_keys);
 	return rc;
 }
+
+int
+dfs_root_chown(daos_handle_t coh, d_string_t user, d_string_t group)
```

Comment:
Couldn't you just dfs_mount/dfs_chmod instead here?  That would be a lot less code to verify and there's no performance consideration here.

Reply:
no because chmod does not handle root directory and for now i don't want it to.
also i don't want to log an error if container is not posix. so this path handles it fine.